crates/evm/fuzz/src/strategies/state.rs

```diff
@@ -159,8 +202,23 @@ impl FuzzDictionary {
     }
 
     #[inline]
-    pub fn values(&self) -> &IndexSet<[u8; 32]> {
-        &self.state_values
+    pub fn values(&self, param: Option<DynSolType>) -> &IndexSet<[u8; 32]> {
+        match param {
+            Some(param_type) => {
+                if let Some(typed_samples) = self.sample_values.get(&param_type) {
+                    // If we have sample values collected for this type then
+                    // use them with a weight of 50.
+                    if rand::thread_rng().gen_range(0..100) < 50 {
+                        typed_samples
```

Comment:
I believe we should use [prop_perturb](https://altsysrq.github.io/rustdoc/proptest/latest/proptest/strategy/trait.Strategy.html#method.prop_perturb) here as randomness source

Reply:
I reorged this part in https://github.com/foundry-rs/foundry/pull/7666/commits/58fc5a91e26bfe7712c85aa3c2060a898a286144 not sure we still need perturb, pls lmk wdyt

Reply:
makes sense, I added a slightly different way using `(0..100).prop_flat_map(Just)` in https://github.com/foundry-rs/foundry/pull/7666/commits/d00c19284e20d78575831735d430016c84c947f9 pls check